h2o-core/src/main/java/water/api/Schema.java

```diff
@@ -513,8 +515,15 @@ public S fillFromAny(Object o) {
     if (KeyV3.class.isAssignableFrom(fclz)) {
       if ((s == null || s.length() == 0) && required) throw new H2OKeyNotFoundArgumentException(field_name, s);
       if (!required && (s == null || s.length() == 0)) return null;
-
-      return KeyV3.make(fclz, Key.make(s.startsWith("\"") ? s.substring(1, s.length() - 1) : s)); // If the key name is in an array we need to trim surrounding quotes.
+      // If the key name is in an array we need to trim surrounding quotes.
+      s = s.startsWith("\"") ? s.substring(1, s.length() - 1) : s;
+      // guess if the s is a JSON encoded value
+      if (s.startsWith("{")) { // PUBDEV-7394
+        // s should contain here just the JSON encoded value ({"value":"xxxx"})
+        Properties parsedS = gson.fromJson(s, Properties.class);
+        s = parsedS.getProperty("value");
+      }
+      return KeyV3.make(fclz, Key.make(s));
```

Comment:
I don't undestand why you have to do this: is it just for Flow? If sth works for other clients, but not for Flow, we should fix it on Flow side.

Reply:
Also, don't like the ad-hoc parsing. If you obtain a json with a value attribute, what is it supposed to be mapped to?

Reply:
I managed to find a place where I can fix it in the flow which would complete remove the need to modify schema.java...

Here is my proposed fix:
```
diff --git a/src/ext/modules/proxy.coffee b/src/ext/modules/proxy.coffee
index 04b3c92d..3375310f 100644
--- a/src/ext/modules/proxy.coffee
+++ b/src/ext/modules/proxy.coffee
@@ -135,6 +135,8 @@ exports.init = (_) ->
           if isNumber element
             return element
           if isObject element
+            if 'value' in element
+              return stringify element.value
             return stringify element
           return "\"#{element}\""
         "[#{mappedArray.join ','}]"
```

The problem is that it's quite hard to test flow and the `element` representing the `KeyV3` is just plain javascript object of type object containing just a single attribute called `value`.

Reply:
I don't think it's the right either